taskvine/src/manager/vine_manager.c

```diff
@@ -5211,6 +5276,9 @@ int vine_cancel_by_task_id(struct vine_manager *q, int task_id)
 		return 0;
 	}
 
+	if (task->type == VINE_TASK_TYPE_LIBRARY_INSTANCE)
```

Comment:
Don't manipulate the refcount directly.
Just call `vine_task_delete` which removes a refcount and will delete if it goes to zero.
(The idea is that we should have mirror pairs of `vine_task_addref` and `vine_task_delete`